volatility3/framework/symbols/windows/extensions/gui.py

```diff
@@ -120,8 +130,193 @@ def get_threads(
 
             yield thread, process_name, process_pid
 
+    def _do_get_windows(
+        self, window, max_windows
+    ) -> Generator[Tuple[interfaces.objects.ObjectInterface, str], None, None]:
+        """
+        Recusively walks and yields the adjacent and child windows
+        """
+        seen_windows = set()
+        seen_children = set()
+
+        if window.vol.offset == 0:
+            return
+
+        yield window, window.get_name()
+
+        seen_windows.add(window)
+
+        # Walk adjacent windows
+        while len(seen_windows) < max_windows:
+            try:
```

Comment:
So this is kind of... isn't breadth first, or depth first?  It's kind of a hybrid?  (Since it doesn't do all nodes on a level, it does all nodes _from that child_ on that level).  Might be better to swap it to simple depth first (children then siblings)?

Reply:
This is matching Vol2 directly. The first loop through `spwndNext` should catch everything on the same level. Then each window's children. Its also written this way so if we ever want to resurrect the screenshot plugin,which draws a PNG of all the windows on the desktop with their titles, then walking the tree this way preserves z-order.

Reply:
Fair enough, but it's not strictly breadth first because children of children will get renderered before... children of sibling's (so it's a localized level, not the whole level of the tree)?  If that's how it worked and that preserves z-order then that's fine, just want to make sure we're walking the tree the right way.